lightly_studio/tests/resolvers/test_twodim_embedding_resolver.py

```diff
@@ -1,13 +1,149 @@
-from lightly_studio.resolvers.twodim_embedding_resolver import _calculate_2d_embeddings
+from __future__ import annotations
+
+import numpy as np
+from pytest_mock import MockerFixture
+from sqlmodel import Session
+
+from lightly_studio.resolvers import twodim_embedding_resolver
+from tests.helpers_resolvers import (
+    create_dataset,
+    create_embedding_model,
+    create_sample,
+    create_sample_embedding,
+)
 
 
 def test__calculate_2d_embeddings__1_sample() -> None:
     embedding_values = [[0.1, 0.2, 0.3]]
-    projected = _calculate_2d_embeddings(embedding_values)
+    projected = twodim_embedding_resolver._calculate_2d_embeddings(embedding_values)
     assert projected == [(0.0, 0.0)]
 
 
 def test__calculate_2d_embeddings__2_samples() -> None:
     embedding_values = [[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]]
-    projected = _calculate_2d_embeddings(embedding_values)
+    projected = twodim_embedding_resolver._calculate_2d_embeddings(embedding_values)
     assert projected == [(0.0, 0.0), (1.0, 1.0)]
+
+
+def test__get_twodim_embeddings__cache_hit(
+    test_db: Session,
+    mocker: MockerFixture,
+) -> None:
+    # Create dataset, embedding model, samples, and embeddings.
+    dataset = create_dataset(session=test_db, dataset_name="cache_hit_dataset")
+    embedding_model = create_embedding_model(
+        session=test_db,
+        dataset_id=dataset.dataset_id,
+        embedding_model_name="cache_hit_model",
+        embedding_model_hash="cache_hit_hash",
+        embedding_dimension=3,
+    )
+
+    embeddings = [
+        [0.1, 0.2, 0.3],
+        [0.4, 0.5, 0.6],
+    ]
+    for index, embedding in enumerate(embeddings):
```

Comment:
Optional: I would prefer not having a for loop in tests which loops only twice. We already have a function to create multiple samples, and `create_sample_embedding` can be just called twice.